sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -226,3 +221,34 @@ export class AzurePowerShellCredential implements TokenCredential {
     });
   }
 }
+
+/**
+ *
+ * @internal
+ */
+export async function parseJsonToken(
+  result: string,
+): Promise<{ Token: string; ExpiresOn: string }> {
+  const jsonRegex = /{[^{}]*}/g;
+  const matches = result.match(jsonRegex);
+  let resultWithoutToken = result;
+  if (matches) {
+    try {
+      for (const item of matches) {
+        try {
+          const jsonContent = JSON.parse(item);
+          if (jsonContent?.Token) {
+            resultWithoutToken = resultWithoutToken.replace(item, "");
+            logger.getToken.warning(resultWithoutToken);
```

Comment:
Should we log warning only when resultWithoutToken is not empty?

Also we assume that there's only one token?  that is, it is not possible that we log sensitive information, since the only one has been removed.

Reply:
No there's only one access token.

Reply:
```suggestion
            if(resultWithoutToken){
                logger.getToken.warning(resultWithoutToken);
            }
```